db/migrate/20170531140414_create_federation_config.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+class CreateFederationConfig < ActiveRecord::Migration
+  def self.up
+    create_table :federation_configs do |t|
+      t.integer :collection_id, null: false # The ID of the collection this config applies to
+      t.string :provider, null: false       # The name of the provider this config applies to
+      t.string :context_query               # The default query which should always be applied.
```

Comment:
I think just "query" would have sufficed for this column name. Collection + provider = context.